lib/ExpensiMark.js

```diff
@@ -169,6 +169,19 @@ export default class ExpensiMark {
                 },
             },
 
+            /**
+             * A room mention is a string that starts with the '#' symbol and is followed by a valid room name.
+             *
+             * Note: We are allowing mentions in a format of #room-name The room name can contain any
+             * combination of letters and hyphens
+             */
+            {
+                name: 'roomMentions',
+
+                regex: /(?<![^ \n*~_])(#[\p{Ll}0-9-]{1,80})/gmiu,
```

Comment:
NAB but certain improvements were introduced as mentioned in [this issue](https://github.com/Expensify/App/issues/46299) with respect to identifying room mentions.